In [52]:
# Import the Pandas library
import pandas as pd
import hvplot.pandas
# Import the Path module from the Pathlib library
from pathlib import Path
#Import numpy
import numpy as np
# Import the visulalization dependencies
%matplotlib inline
#import hvplot.pandas

In [53]:
# Using the read_csv function and the Path module, read in the "WB_DATA_METADATA.csv" file from the data folder for series names .
# Set the `year` as the index column. 

wb_metadata = pd.read_csv(
    Path('../data\WB_METADATA.csv'),  
)

# Review the DataFrame with both the head & tail functions
wb_metadata


series_id                                               name  \
0     AG.LND.IRIG.AG.ZS  Agricultural irrigated land (% of total agricu...   
1        AG.LND.AGRI.ZS                 Agricultural land (% of land area)   
2        AG.LND.AGRI.K2                         Agricultural land (sq. km)   
3        SH.DYN.AIDS.ZS  Prevalence of HIV, total (% of population ages...   
4        DC.DAC.CANL.CD  Net bilateral aid flows from DAC donors, Canad...   
...                 ...                                                ...   
1442     SL.UEM.BASC.ZS  Unemployment with basic education (% of total ...   
1443  SL.UEM.BASC.FE.ZS  Unemployment with basic education, female (% o...   
1444  SL.UEM.BASC.MA.ZS  Unemployment with basic education, male (% of ...   
1445     SL.UEM.INTM.ZS  Unemployment with intermediate education (% of...   
1446  SL.UEM.INTM.FE.ZS  Unemployment with intermediate education, fema...   

                                            description  
0     Agricultural irrigated land refers to agricult...  
1     Agricultural land refers to the share of land ...  
2     Agricultural land refers to the share of land ...  
3     Prevalence of HIV refers to the percentage of ...  
4     Net bilateral aid flows from DAC donors are th...  
...                                                 ...  
1442  The percentage of the labor force with a basic...  
1443  The percentage of the labor force with a basic...  
1444  The percentage of the labor force with a basic...  
1445  The percentage of the labor force with an inte...  
1446  The percentage of the labor force with an inte...  

[1447 rows x 3 columns]

# Data Collection Stage 
Step 1: Create dataframes for each of the seven countries selected for analysis with all the series.

In [54]:
# Using the read_csv function and the Path module, to read various country csv files from the data folder.
# Set the `year` as the index column. 

wb_australia = pd.read_csv(
    Path('../data\WB_DATA_AUS.csv'), index_col='year'
    )
wb_brazil = pd.read_csv(
    Path('../data\WB_DATA_BRA.csv'), index_col='year'
    )
wb_china= pd.read_csv(
    Path('../data\WB_DATA_CHN.csv'),  index_col='year' 
    )
wb_germany= pd.read_csv(
    Path('../data\WB_DATA_DEU.csv'), index_col='year' 
    )
wb_nigeria= pd.read_csv(
    Path('../data\WB_DATA_NGA.csv'), index_col='year'
    )
wb_saudi= pd.read_csv(
    Path('../data\WB_DATA_SAU.csv'),  index_col='year' 
    )
wb_usa= pd.read_csv(
    Path('../data\WB_DATA_USA.csv'), index_col='year'
    )
# Review the DataFrame with both the head & tail functions
wb_australia


series_id country_code country_name         value
year                                                        
1990  NV.AGR.TOTL.ZS          AUS    Australia  4.194135e+00
1991  NV.AGR.TOTL.ZS          AUS    Australia  3.171618e+00
1992  NV.AGR.TOTL.ZS          AUS    Australia  3.043311e+00
1993  NV.AGR.TOTL.ZS          AUS    Australia  3.253233e+00
1994  NV.AGR.TOTL.ZS          AUS    Australia  3.338470e+00
...              ...          ...          ...           ...
2017     EN.URB.MCTY          AUS    Australia  1.492426e+07
2018     EN.URB.MCTY          AUS    Australia  1.521236e+07
2019     EN.URB.MCTY          AUS    Australia  1.544669e+07
2020     EN.URB.MCTY          AUS    Australia  1.567826e+07
2021     EN.URB.MCTY          AUS    Australia  1.590534e+07

[38733 rows x 4 columns]

# Agriculture Land Area Series Analysis 
## 1. Extract the  AG.LND.AGRI.ZS series from the country dataframe for analysis.This series contains percentage of agricultural land area in a country. Agricultural land refers to the share of land area that is arable, under permanent crops, and under permanent pastures.

In [55]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country  
Australia_agri_df = wb_australia [wb_australia ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

Australia_agri_df.rename(columns={'value':'Australia'},inplace=True)
Australia_agri_df.sort_index
Australia_agri_df.tail()

Australia
year           
2014  48.738399
2015  45.314424
2016  44.539926
2017  48.340618
2018  46.658095

In [56]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country  

Brazil_agri_df = wb_brazil [wb_brazil ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

Brazil_agri_df.rename(columns={'value':'Brazil'},inplace=True)

Brazil_agri_df.tail()

Brazil
year           
2014  28.082492
2015  28.168731
2016  28.254971
2017  28.341066
2018  28.341090

In [57]:
Brazil_agri_df.describe()

Brazil
count  60.000000
mean   26.274024
std     2.801890
min    18.779178
25%    25.310625
50%    27.386057
75%    28.087954
max    28.953656

In [58]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 

China_agri_df = wb_china [wb_china ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

China_agri_df.rename(columns={'value':'China'},inplace=True)
China_agri_df.sort_index
China_agri_df.tail()

China
year           
1995  55.568123
1996  55.572214
1997  55.571392
1998  55.570781
1999  55.570384

In [59]:
China_agri_df.describe()

China
count  60.000000
mean   49.766309
std     7.185868
min    36.420045
25%    43.084627
50%    54.052980
75%    55.898829
max    56.133200

In [60]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 
Germany_agri_df = wb_germany [wb_germany ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

Germany_agri_df.rename(columns={'value':'Germany'},inplace=True)
Germany_agri_df.sort_index
Germany_agri_df.tail()

Germany
year           
2014  47.936371
2015  47.959067
2016  47.684337
2017  47.763116
2018  47.641536

In [61]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 
Nigeria_agri_df = wb_nigeria [wb_nigeria ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

# Brazil_df.drop(columns=['series_id','country_code'])
Nigeria_agri_df .rename(columns={'value':'Nigeria'},inplace=True)
Nigeria_agri_df.sort_index
Nigeria_agri_df.tail()

Nigeria
year           
2014  75.178414
2015  75.357719
2017  75.716317
2018  75.895616
1978  53.751221

In [62]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 
Saudi_agri_df = wb_saudi [wb_saudi ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

Saudi_agri_df.rename(columns={'value':'Saudi'},inplace=True)
Saudi_agri_df.sort_index
Saudi_agri_df.tail()

Saudi
year           
2000  80.841889
2001  80.844680
2002  80.845610
2003  80.847936
2004  80.806535

In [63]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 
USA_agri_df = wb_usa [wb_usa ['series_id']=='AG.LND.AGRI.ZS'].drop(['country_code','country_name','series_id'], axis=1)

# Brazil_df.drop(columns=['series_id','country_code'])
USA_agri_df.rename(columns={'value':'USA'},inplace=True)
USA_agri_df.sort_index
USA_agri_df.tail()

USA
year           
2016  44.303705
2017  44.363367
2018  44.363367
2019  44.363367
2020  44.363367

## 2. Data Preparation  Stage
##  Join the  dataframes for the seven countries. Clean the data by dropping any NaNs, duplicates 

In [64]:

# Concatenate the country DataFrames

wb_countries_agri = pd.concat([Australia_agri_df, Brazil_agri_df, China_agri_df, Germany_agri_df, Nigeria_agri_df,  Saudi_agri_df, USA_agri_df], axis = 1).sort_values(by = 'year')
# Review the  DataFrame 
wb_countries_agri.dtypes



Australia    float64
Brazil       float64
China        float64
Germany      float64
Nigeria      float64
Saudi        float64
USA          float64
dtype: object

In [65]:
# Check  for missing data
wb_countries_agri.isnull()


Australia  Brazil  China  Germany  Nigeria  Saudi    USA
year                                                          
1961      False   False  False    False    False  False  False
1962      False   False  False    False    False  False  False
1963      False   False  False    False    False  False  False
1964      False   False  False    False    False  False  False
1965      False   False  False    False    False  False  False
1966      False   False  False    False    False  False  False
1967      False   False  False    False    False  False  False
1968      False   False  False    False    False  False  False
1969      False   False  False    False    False  False  False
1970      False   False  False    False    False  False  False
1971      False   False  False    False    False  False  False
1972      False   False  False    False    False  False  False
1973      False   False  False    False    False  False  False
1974      False   False  False    False    False  False  False
1975      False   False  False    False    False  False  False
1976      False   False  False    False    False  False  False
1977      False   False  False    False    False  False  False
1978      False   False  False    False    False  False  False
1979      False   False  False    False    False  False  False
1980      False   False  False    False    False  False  False
1981      False   False  False    False    False  False  False
1982      False   False  False    False    False  False  False
1983      False   False  False    False    False  False  False
1984      False   False  False    False    False  False  False
1985      False   False  False    False    False  False  False
1986      False   False  False    False    False  False  False
1987      False   False  False    False    False  False  False
1988      False   False  False    False    False  False  False
1989      False   False  False    False    False  False  False
1990      False   False  False    False    False  False  False
1991      False   False  False    False    False  False  False
1992      False   False  False    False    False  False  False
1993      False   False  False    False    False  False  False
1994      False   False  False    False    False  False  False
1995      False   False  False    False    False  False  False
1996      False   False  False    False    False  False  False
1997      False   False  False    False    False  False  False
1998      False   False  False    False    False  False  False
1999      False   False  False    False    False  False  False
2000      False   False  False    False    False  False  False
2001      False   False  False    False    False  False  False
2002      False   False  False    False    False  False  False
2003      False   False  False    False    False  False  False
2004      False   False  False    False    False  False  False
2005      False   False  False    False    False  False  False
2006      False   False  False    False    False  False  False
2007      False   False  False    False    False  False  False
2008      False   False  False    False    False  False  False
2009      False   False  False    False    False  False  False
2010      False   False  False    False    False  False  False
2011      False   False  False    False    False  False  False
2012      False   False  False    False    False  False  False
2013      False   False  False    False    False  False  False
2014      False   False  False    False    False  False  False
2015      False   False  False    False    False  False  False
2016      False   False  False    False    False  False  False
2017      False   False  False    False    False  False  False
2018      False   False  False    False    False  False  False
2019      False   False  False    False    False  False  False
2020      False   False  False    False    False  False  False

In [66]:
wb_countries_agri.isnull().sum()

Australia    0
Brazil       0
China        0
Germany      0
Nigeria      0
Saudi        0
USA          0
dtype: int64

In [67]:
# Check for duplicates 
wb_countries_agri.duplicated().sum()

0

In [68]:
wb_countries_agri.head(10)

Australia     Brazil      China    Germany    Nigeria      Saudi  \
year                                                                     
1961  61.763332  18.779178  36.420045  55.495088  58.386321  40.084849   
1962  62.037792  19.222698  36.712150  55.546645  59.180693  40.094153   
1963  62.399227  19.666218  37.014865  55.572423  61.505001  40.104387   
1964  62.729106  20.109737  37.317474  55.732822  62.552642  40.117877   
1965  63.296043  20.553257  37.720988  55.950506  64.857351  40.127646   
1966  63.396187  20.996777  38.103282  55.941913  64.551424  40.137880   
1967  63.728384  21.440297  38.485575  55.601066  63.916104  40.147184   
1968  66.851016  21.883816  38.867338  55.237304  63.871175  40.163931   
1969  66.501962  22.327336  39.357009  55.148512  70.066021  40.180212   
1970  64.229788  22.771574  39.844027  54.486867  67.009750  40.201145   

            USA  
year             
1961  48.860242  
1962  48.476137  
1963  48.197940  
1964  47.932516  
1965  47.589792  
1966  47.554526  
1967  47.527994  
1968  47.581276  
1969  47.543280  
1970  47.428966

## 3 Agriculture Land Area Data Analysis

In [69]:

 #Visualize the perctage agricultural land area  using the hvplot function
wb_countries_agri.hvplot.line(
xlabel = 'Year', ylabel = '% agricultural land area',frame_width =800,
    frame_height = 400,
title="Agric Land  Area ")


:NdOverlay   [Variable]
   :Curve   [year]   (value)

# Observations
* Based on the visualisations above , Saudi Arabia had the highest jump in the percentage of agricultural land area between 1983 and 1999 doubling from 40.6%  tp 80.8%
* For Nigeria the period between 1970 and 1981 registered a decline from 70.06% to 51.8% . The trend revereved between 1982 and 1992 when there was an increase to 70.7% followed by a period of steady growth . China and Brazil also registered growth in the period under review  and their growth  exhibited similar patterns, this relationship will be further explored below. 
* Germany and United States registered a decline in the percentage share in the agricultural land area during the perod of the review. 

# Further Analysis of China and Brazil  Data 
## Using the 5-year rolling window, the agric land area data for  China and Brazil s, calculate the covariance of  Agric Land Area allocation for the two countries.

In [70]:
#Create a new dataframe of the convariance between  China and Brazil's  agric land area and food productiion using a 5 year rolling window. 
china_brazil_rolling_agric_convarince= wb_countries_agri['China'].rolling(window=5).cov( wb_countries_agri['Brazil']).dropna()

china_brazil_rolling_agric_convarince.tail()

year
2016   -0.001513
2017   -0.001648
2018   -0.000963
2019   -0.000433
2020   -0.000079
dtype: float64

In [71]:
#Visualize the convarinance for China  and Brazil  agric land area
china_brazil_rolling_agric_convarince.hvplot.line(label="China_Brazil_Agric_Land Area Convariance",color="blue",frame_width =800,
    frame_height = 400) 

:Curve   [year]   (0)

There is a strong positive relationship of the agriculrure land area  for China and Brazil in the perriod 1965 to 1986 meaning that they tended to similarly move up and down on a yearly basis. After 1986 this relationship is weak. 
 
This may imply either country may have adopted different policy direction on agriacultural land area in the latter years of the analyisis. 

# Part 2:  Food Production  Analysis per Country

## DATA COLLECTION STAGE

In [72]:
#  Extracting AG.PRD.FOOD.XD series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the food production index (2014-2016 = 100) a country  
Australia_food_df = wb_australia [wb_australia ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

Australia_food_df.rename(columns={'value':'Australia'},inplace=True)
Australia_food_df.sort_index
Australia_food_df.tail()

Australia
year           
2016      98.22
2017     108.34
2018      97.37
2019      91.35
2020      89.00

In [73]:
Brazil_food_df = wb_brazil [wb_brazil ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

Brazil_food_df.rename(columns={'value':'Brazil'},inplace=True)

Brazil_food_df.tail()

Brazil
year        
2018  107.09
2019  109.26
2020  111.85
1983   29.95
1984   31.89

In [74]:
China_food_df = wb_china [wb_china ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

China_food_df.rename(columns={'value':'China'},inplace=True)
China_food_df.sort_index
China_food_df.tail()

China
year        
2016  100.86
2017  102.59
2018  103.25
2019  101.90
2020  102.96

In [75]:
Germany_food_df = wb_germany [wb_germany ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

Germany_food_df .rename(columns={'value':'Germany'},inplace=True)
Germany_food_df .sort_index
Germany_food_df .tail()

Germany
year         
2017    97.50
2018    91.58
2019    93.67
2020    94.74
2008    96.41

In [76]:
Nigeria_food_df = wb_nigeria [wb_nigeria ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

# Brazil_df.drop(columns=['series_id','country_code'])
Nigeria_food_df .rename(columns={'value':'Nigeria'},inplace=True)
Nigeria_food_df.sort_index
Nigeria_food_df.tail()

Nigeria
year         
2016   104.49
2017   105.25
2018   104.69
2019   106.46
2020   106.31

In [77]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 
Saudi_food_df = wb_saudi [wb_saudi ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

Saudi_food_df .rename(columns={'value':'Saudi'},inplace=True)
Saudi_food_df .sort_index
Saudi_food_df .tail()

Saudi
year        
2016  106.72
2017  112.24
2018  126.20
2019  137.66
2020  151.14

In [78]:
#  Extracting AG.LND.AGRI.ZS series from the country dataframe for analysis 
#  Create  a new dataframe for the country with 'value' colunm for the percentage of areas land in a country 
USA_food_df = wb_usa [wb_usa ['series_id']=='AG.PRD.FOOD.XD'].drop(['country_code','country_name','series_id'], axis=1)

# Brazil_df.drop(columns=['series_id','country_code'])
USA_food_df.rename(columns={'value':'USA'},inplace=True)
USA_food_df.sort_index
USA_food_df.tail()

USA
year        
2018  103.35
2019  100.09
2020  104.45
2002   80.45
1969   49.84

## DATA PREPARATION STAGE
## Join the  dataframes for the seven countries. Clean the data by dropping any NaNs, duplicates 

In [79]:
# Concatenate the country DataFrames

wb_countries_food = pd.concat([Australia_food_df, Brazil_food_df, China_food_df, Germany_food_df, Nigeria_food_df,  Saudi_food_df, USA_food_df], axis = 1).sort_values(by = 'year')
# Review the  DataFrame
wb_countries_agri.dtypes

Australia    float64
Brazil       float64
China        float64
Germany      float64
Nigeria      float64
Saudi        float64
USA          float64
dtype: object

In [80]:
# Check  for NaNs
wb_countries_food.isnull().sum()

Australia    0
Brazil       0
China        0
Germany      0
Nigeria      0
Saudi        0
USA          0
dtype: int64

In [81]:
# Check for duplicates 
wb_countries_food.duplicated().sum()

0

In [82]:
#Review the combined countries food index dataframe
wb_countries_food.head(10)

Australia  Brazil  China  Germany  Nigeria  Saudi    USA
year                                                          
1961      32.31   12.22  11.43    71.93    18.32  10.89  41.26
1962      36.69   12.71  12.10    78.87    19.26  10.69  41.32
1963      38.42   13.25  12.63    82.32    20.17  10.83  42.95
1964      40.78   13.84  13.40    80.71    20.75  11.06  43.50
1965      39.31   15.29  14.56    77.82    21.64  11.73  45.36
1966      43.14   14.99  15.63    81.82    20.76  11.82  46.32
1967      39.15   15.98  15.84    88.16    22.07  12.52  48.67
1968      45.21   16.69  15.77    88.79    22.53  10.94  49.21
1969      42.95   17.16  16.06    85.24    24.99  13.04  49.84
1970      45.26   18.09  17.02    88.35    26.75  14.36  49.13

##  FOOD PRODUCTION DATA ANALYSIS

In [83]:

# Visualize the food production index  using the plot function
wb_countries_food.hvplot.line(
xlabel = 'Year', ylabel = 'Food Production Index (2014-2016=100)',frame_width =800,
    frame_height = 400,
title="Country LevelFood Prduction Index ")

:NdOverlay   [Variable]
   :Curve   [year]   (value)

# AGRICULTURE LAND AREA AND  FOOD PRODUCTION  COMBINED ANALYSIS
##  Calculate convarinaces of the food production and agriculture land area per country
## Visualize the matrices for each contry 

In [84]:
# Calculate the convariance bettween the agriculture land area per country and food production levels 
china_agric_food_convariance= wb_countries_agri['China'].rolling(window=5).cov( wb_countries_food['China']).dropna()

china_agric_food_convariance.tail()

year
2016   -0.030773
2017   -0.029726
2018   -0.015930
2019   -0.003860
2020   -0.001673
Name: China, dtype: float64

In [85]:
# Calculate the convariance bettween the agriculture land area per country and food production levels 
brazil_agric_food_convariance= wb_countries_agri['Brazil'].rolling(window=5).cov( wb_countries_food['Brazil']).dropna()

brazil_agric_food_convariance.tail()

year
2016    0.427316
2017    0.408120
2018    0.366063
2019    0.257403
2020    0.169008
Name: Brazil, dtype: float64

In [86]:
# Calculate the convariance bettween the agriculture land area per country and food production levels 
Australia_agric_food_convariance= wb_countries_agri['Australia'].rolling(window=5).cov( wb_countries_food['Australia']).dropna()

Australia_agric_food_convariance.tail()

year
2016   -3.147986
2017    2.510663
2018    4.352366
2019    2.839042
2020    4.263851
Name: Australia, dtype: float64

In [87]:
# Calculate the convariance bettween the agriculture land area per country and food production levels 
Germany_agric_food_convariance= wb_countries_agri['Germany'].rolling(window=5).cov( wb_countries_food['Germany']).dropna()

Germany_agric_food_convariance.tail()

year
2016    0.168145
2017    0.171802
2018    0.485113
2019    0.295736
2020    0.094396
Name: Germany, dtype: float64

In [88]:
# Review relationship bettween the agriculture land area per country and food production levels 
Nigeria_agric_food_convariance= wb_countries_agri['Nigeria'].rolling(window=5).cov( wb_countries_food['Nigeria']).dropna()

Nigeria_agric_food_convariance.tail()

year
2016    2.092932
2017    2.137739
2018    1.010359
2019    0.684025
2020    0.217399
Name: Nigeria, dtype: float64

In [89]:
# Review relationship bettween the agriculture land area per country and food production levels 
Saudi_agric_food_convariance= wb_countries_agri['Saudi'].rolling(window=5).cov( wb_countries_food['Saudi']).dropna()

Saudi_agric_food_convariance.tail()

year
2016    0.063116
2017    0.011268
2018    0.003005
2019   -0.111085
2020   -0.072813
Name: Saudi, dtype: float64

In [90]:
# Review relationship bettween the agriculture land area per country and food production levels 
USA_agric_food_convariance= wb_countries_agri['USA'].rolling(window=5).cov( wb_countries_food['USA']).dropna()

USA_agric_food_convariance.tail()

year
2016    0.432804
2017    0.282104
2018    0.194679
2019    0.082172
2020   -0.028518
Name: USA, dtype: float64

In [91]:
#Create new dataframe joining the  agriculture land area and food production levels per country


wb_countries_agri_food_combined = pd.concat([wb_countries_food, wb_countries_agri,], axis = 1).sort_values(by = 'year')
# Review the  DataFrame 
wb_countries_agri_food_combined.head()
column_names=['Australia_Food','Brazil_Food','China_Food','Germany_Food','Nigeria_Food','Saudi_Food','USA_Food','Australia_Agri','Brazil_Agri','China_Agri','Germany_Agri','Nigeria_Agri','Saudi_Agri','USA_Agri']
wb_countries_agri_food_combined.columns = column_names

In [92]:
# Visualize relationship between  share of agrictulture land area  vs  food production level per Country  with plot  and the convariance of the two matrices.
(wb_countries_agri_food_combined[['Australia_Food']].hvplot.line(label="Australia_Food_Production_Index",color="blue",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['Australia_Agri']].hvplot.line(label="Australia Agric Land Area Percentage",color="orange")) + (Australia_agric_food_convariance.hvplot( title="Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I       :Overlay
      .Curve.Australia_Food_Production_Index      :Curve   [year]   (Australia_Food)
      .Curve.Australia_Agric_Land_Area_Percentage :Curve   [year]   (Australia_Agri)
   .Curve.Australia :Curve   [year]   (Australia)

This plot shows that the magnitude of the changes in Australia's 5 year rolling daily food production values vs. the market agriculture land area did not  remain constant over time. Specifically, after 2003 there  is no apparent relationship between the agriculture land area and the food production levels.

In [93]:
# Visualize relationship between  share of agrictulture land area  vs  food production level per Country  with plot  and the convariance of the two matrices.
(wb_countries_agri_food_combined[['Brazil_Food']].hvplot.line(label="Brazil_Food_Production_Index",color="green",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['Brazil_Agri']].hvplot.line(label="Brazil Agric Land Area Percentage",color="red")) + (brazil_agric_food_convariance.hvplot(title="Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I    :Overlay
      .Curve.Brazil_Food_Production_Index      :Curve   [year]   (Brazil_Food)
      .Curve.Brazil_Agric_Land_Area_Percentage :Curve   [year]   (Brazil_Agri)
   .Curve.Brazil :Curve   [year]   (Brazil)

There was strong positive correlationship between the agriculture land area and the food production levels for Brazi to  around 1985 but this trend reversed thereafter and the there is food production increase during the later years of the review although percentage of agriculture land was constant. 

In [94]:
# Visualize relationship between  share of agrictulture land area  vs  food production level per Country  with plot  and the convariance of the two matrices.
(wb_countries_agri_food_combined[['China_Food']].hvplot.line(label="China_Food_Production_Index",color="blue",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['China_Agri']].hvplot.line(label="Agric Land Area Percentage",color="orange")) + (china_agric_food_convariance.hvplot(title="China Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I   :Overlay
      .Curve.China_Food_Production_Index :Curve   [year]   (China_Food)
      .Curve.Agric_Land_Area_Percentage  :Curve   [year]   (China_Agri)
   .Curve.China :Curve   [year]   (China)


There was some  strong  positive correlation between the agriculture land area and the food production levels for Brazil up to 1997 but the correlation was weaker  during the later years of the review.

In [95]:
# Visualize relationship between  share of agrictulture land area  vs  Food Production level per Country
(wb_countries_agri_food_combined[['Germany_Food']].hvplot.line(label="Germany_Food_Production_Index",color="purple",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['Germany_Agri']].hvplot.line(label="Germany Agric Land Area Percentage",color="orange")) + (Germany_agric_food_convariance.hvplot(title="Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I     :Overlay
      .Curve.Germany_Food_Production_Index      :Curve   [year]   (Germany_Food)
      .Curve.Germany_Agric_Land_Area_Percentage :Curve   [year]   (Germany_Agri)
   .Curve.Germany :Curve   [year]   (Germany)

Except for 1992  which showed a strong positive correlation between the two metrices, Germany has exhibited a weak negaive relatioship between agricuture land area and food production levels during review period

In [96]:
# Visualize relationship between  share of agrictulture land area  vs  food production level per Country  with plot  and the convariance of the two matrices.
(wb_countries_agri_food_combined[['Nigeria_Food']].hvplot.line(label="Nigeria_Food_Production_Index",color="green",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['Nigeria_Agri']].hvplot.line(label="Nigeria Agric Land Area Percentage",color="indigo")) + (Nigeria_agric_food_convariance.hvplot(title="Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I     :Overlay
      .Curve.Nigeria_Food_Production_Index      :Curve   [year]   (Nigeria_Food)
      .Curve.Nigeria_Agric_Land_Area_Percentage :Curve   [year]   (Nigeria_Agri)
   .Curve.Nigeria :Curve   [year]   (Nigeria)

There is largely a strong positive linear relationship between Food production value and agriculture land area for Nigeria.  Food production index was below the agriculture land area up 2003 when this trend reversed. 


In [97]:
# Visualize relationship between  share of agrictulture land area  vs  food production level per Country  with plot  and the convariance of the two matrices.
(wb_countries_agri_food_combined[['Saudi_Food']].hvplot.line(label="Saudi_Food_Production_Index",color="blue",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['Saudi_Agri']].hvplot.line(label="Saudi Agric Land Area Percentage",color="orange")) + (Saudi_agric_food_convariance.hvplot( title="Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I   :Overlay
      .Curve.Saudi_Food_Production_Index      :Curve   [year]   (Saudi_Food)
      .Curve.Saudi_Agric_Land_Area_Percentage :Curve   [year]   (Saudi_Agri)
   .Curve.Saudi :Curve   [year]   (Saudi)

For Saudi, apart from the period 1987 to 1994 where there a strong positive convatiance between the two metrices, agriculture land area and food production, the relationship is  generally weak.

In [98]:
# Visualize relationship between  share of agrictulture land area  vs  food production level per Country  with plot  and the convariance of the two matrices.
(wb_countries_agri_food_combined[['USA_Food']].hvplot.line(label="USA_Food_Production_Index",color="green",frame_width =800,
    frame_height = 400) * wb_countries_agri_food_combined[['USA_Agri']].hvplot.line(label="USA Agric Land Area Percentage",color="red")) + (USA_agric_food_convariance.hvplot( title="Agric Land Area and Food Production Convariance"))


:Layout
   .Overlay.I :Overlay
      .Curve.USA_Food_Production_Index      :Curve   [year]   (USA_Food)
      .Curve.USA_Agric_Land_Area_Percentage :Curve   [year]   (USA_Agri)
   .Curve.USA :Curve   [year]   (USA)

 For USA , there has been  weak negative correlation between the agric land area and the food production levels 